### PR TITLE
store/tikv: fix retry open store

### PR DESCRIPTION
**Pull request by nolouch**
Ref https://github.com/pingcap/pd/pull/391 now PD restart spend more time to get the leader.

`kv.IsRetryableError` not work for  the error return by `driver.Open` , I think we can simply let it be true.

```
2016/11/29 11:27:22 printer.go:34: [info] UTC Build Time:  2016-11-29 01:59:37
2016/11/29 11:27:22 client.go:52: [info] [pd] create pd client with endpoints [127.0.0.1:52379]
2016/11/29 11:27:22 conn.go:67: [warning] failed to get leader from [http://127.0.0.1:52379] 
2016/11/29 11:27:23 main.go:155: [fatal] github.com/pingcap/pd/pd-client/rpc_worker.go:361: [pd] rpc failed: read tcp 127.0.0.1:35656->127.0.0.1:52379: i/o timeout
github.com/pingcap/pd/pd-client/rpc_worker.go:249: 
github.com/pingcap/pd/pd-client/rpc_worker.go:227: 
github.com/pingcap/pd/pd-client/rpc_worker.go:77: 
github.com/pingcap/pd/pd-client/client.go:55: 
/data/jenkins/workspace/BENCH_TEST_ON_UCLOUD/go/src/github.com/pingcap/tidb/store/tikv/kv.go:59: 
/data/jenkins/workspace/BENCH_TEST_ON_UCLOUD/go/src/github.com/pingcap/tidb/tidb.go:220:

### CODE REVIEW

**Comment by coocood** [comment]
This is not the right way to fix this issue, we should check the particular error and add it in `kv.IsRetryableErr`.

**Comment by nolouch** [comment]
PTAL @coocood

File: store/tikv/kv.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ func (d Driver) Open(path string) (kv.Storage, error) {
 
 	pdCli, err := pd.NewClient(etcdAddrs)
 	if err != nil {
-		return nil, errors.Trace(err)
+		return nil, errors.Annotate(errors.Trace(err), "try again later")
```
Comment (coocood) [inline store/tikv/kv.go]: What if this error is actually not retryable, like malformed `etcdAddrss`?

**Comment by coocood** [comment]
LGTM

File: store/tikv/kv.go [hunk 2]
```diff
@@ -56,6 +56,9 @@ func (d Driver) Open(path string) (kv.Storage, error) {
 
 	pdCli, err := pd.NewClient(etcdAddrs)
 	if err != nil {
+		if strings.Contains(err.Error(), "i/o timeout") {
+			return nil, errors.Annotate(errors.Trace(err), "try again later")
```
Comment (zimulala) [inline store/tikv/kv.go]: Define `try again later` as a constant.

**Comment by tiancaiamao** [comment]
@disksing

File: store/tikv/kv.go (hunk 2 above)
Comment (disksing) [inline store/tikv/kv.go]: We have a const `txnRetryableMark` inside the package.

Comment (disksing) [inline store/tikv/kv.go]: BTW, when use `errors.Annotate`, you don't need to `errors.Trace`.

**Comment by nolouch** [comment]
PTAL @zimulala @disksing

**Comment by disksing** [comment]
LGTM

**Pull request merged**